### DISCUSSION TITLE
Waiting Stratergy

### DISCUSSION THREAD

**Discussion by shameer491998** [selenide/selenide#2814] [Q&A]
I have a test case for initiating a transaction while selecting from account from a drop down, a new window pops up, upon selection the added account gets selected as from account automatically. After selection a loading skeleton appears which causes the page to refresh . The next step is to select the to account. The from and to account gets selected even before the loading skeleton appears. But when the loading skeleton finally appears after the selection of accounts, the page refreshes which results in the to account to get deselected due to which the the script breaks. I have tried multiple ways including waiting for the skeleton to disappear. Can anyone guide me on how to wait for the entire page to be in ready state or complete  before the other input values can be entered? @asolntsev

**Comment by anilreddy** [comment]
```java
import static com.codeborne.selenide.Condition.hidden;
import static com.codeborne.selenide.Condition.visible;

$(element).shouldBe(visible) or $(element).shouldBe(hidden)

```

@shameer491998 as per above code you can identify last element in page that loads and use wait for it or else you can use please wait loading mechanism to be hidden anything would work.

**Reply by shameer491998** [reply]
Thanks for your inputs. Could you elaborate on please wait loading mechanism?

**Reply by anilreddy** [reply]
That should be known in your page how can i elaborate?
